docs: add metadata to all remaining pages (#1246)
canonical/rockcraft: docs/reference/plugins.rst

.. meta::

>     :description: Reference documentation for all Rockcraft plugins for all supported languages and frameworks.

Reviewer: ```suggestion
    :description: The reference documentation for all part plugins in Rockcraft.
```
